internal/util/diff/diff.go

```diff
@@ -102,34 +103,44 @@ func (c *Command) Run() error {
 		return errors.Errorf("package missing Kptfile at '%s': %v", c.Path, err)
 	}
 
-	// Stage current package
-	currPkg, err := ioutil.TempDir("", "kpt-")
+	// Create a staging directory to store all compared packages
+	stagingDirectory, err := ioutil.TempDir("", "kpt-")
 	if err != nil {
-		return errors.Errorf("failed to create stage dir for current package: %v", err)
+		return errors.Errorf("failed to create stage dir: %v", err)
 	}
 	defer func() {
+		// Cleanup staged content after diff. Ignore cleanup if debugging.
 		if !c.Debug {
-			defer os.RemoveAll(currPkg)
+			defer os.RemoveAll(stagingDirectory)
 		}
 	}()
 
+	// Stage current package
+	// This prevents prepareForDiff from modifying the local package
+	currPkg, err := stageDirectory(stagingDirectory, "local-package")
+	if err != nil {
+		return errors.Errorf("failed to create stage dir for current package: %v", err)
+	}
+
 	err = copyutil.CopyDir(c.Path, currPkg)
 	if err != nil {
 		return errors.Errorf("failed to stage current package: %v", err)
 	}
+	fmt.Printf("Staging %s at %s\n", c.Path, currPkg)
```

Comment:
is this the correct ordering of arguments.  Seems like: Staging a <package> at <directory>?

Reply:
This will print: `Staging dir at /tmp/local-package`. The format I have used is "Staging `some-source` at `some-staging-location`"

Reply:
got it.  The variable naming is a bit misleading but it seems like it's printing the right stuff.  Thanks for clarifying!